tests/src/testcommon.hh

```diff
@@ -187,139 +186,88 @@ template <typename NonLinearOperator>
-  auto gridView        = fe.localView().globalBasis().gridView();
-  auto scalarBasis     = makeConstSharedBasis(gridView, lagrangeDG<1>());
-  auto localScalarView = scalarBasis->localView();
-  std::vector<Dune::FieldVector<double, 3>> stressVector(scalarBasis->size());
-
-  auto ele = elements(gridView).begin();
-  localScalarView.bind(*ele);
-  const auto& fe2              = localScalarView.tree().finiteElement();
-  const auto& referenceElement = Dune::ReferenceElements<double, gridDim>::general(ele->type());
-  for (const auto c : std::views::iota(0u, fe2.size())) {
-    const auto fineKey                        = fe2.localCoefficients().localKey(c);
-    const auto nodalPositionInChildCoordinate = referenceElement.position(fineKey.subEntity(), fineKey.codim());
-    fe.calculateAt(resultRequirements, nodalPositionInChildCoordinate, result);
-    Eigen::Vector3d computedResult = result.getResult(ResultType::linearStress);
-    const auto nodeIndex           = localScalarView.index(localScalarView.tree().localIndex(c))[0];
-    stressVector[nodeIndex]        = toDune(computedResult);
-    for (const auto voigtIndex : std::views::iota(0u, static_cast<unsigned int>(gridDim * (gridDim + 1) / 2))) {
-      const auto FEStressComponent       = stressVector[nodeIndex][voigtIndex];
-      const auto expectedStressComponent = expectedStress[c][voigtIndex];
-      const bool isStressCorrect
-          = Dune::FloatCmp::eq<double, Dune::FloatCmp::CmpStyle::absolute>(FEStressComponent, expectedStressComponent);
-      t.check(isStressCorrect) << "Stress component " << voigtIndex << " at node " << c << " is not correctly computed"
-                               << messageIfFailed;
-    }
-  }
-  return t;
-}
-
-template <typename NonLinearOperator, typename FiniteElement>
-[[nodiscard]] auto checkLinearStressOfSimplex(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
-                                              const std::string& messageIfFailed = "") {
+[[nodiscard]] auto checkLinearStress(NonLinearOperator& nonLinearOperator, FiniteElement& fe,
+                                     const std::string& messageIfFailed = "") {
   constexpr int dim = FiniteElement::Traits::mydim;
 
   static_assert(std::remove_cvref_t<decltype(fe.localView().tree())>::degree() == dim,
-                "The test to check linear stress is only supported with the powerBasis being two-dimensional");
-  assert(fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
-         && "The test to check linear stress is only supported for simplices");
+                "The test to check linear stress is only supported with the powerBasis having the same power as the "
+                "element dimension");
+  assert(fe.localView().element().type() == Dune::GeometryTypes::cube(dim)
+         or fe.localView().element().type() == Dune::GeometryTypes::simplex(dim)
+                && "The test to check linear stress is only supported for cubes");
```

Comment:
The assert message is out of sync.

Reply:
Ah yeah sorry, I was mindlessly copy-pasting these